### DISCUSSION TITLE
is it possible to have different table for list and different table for form

### DISCUSSION THREAD

**Discussion by SledgehammerPL** [wintercms/winter#797] [Q&A] [answered]
in model definition there's a field:

 public $table = 'my_plugin_table1';

But I prefer to have views which are much more flexible to show columns joined and filtered on database side

But of course - you can't easy edit view (in postgresql it is possible but not quite easy)

so i would like to have two definitions:

 public $list_table = 'my_plugin_table1_view';
 public $form_table = 'my_plugin_table1';

of course most users would have exact the same values in both - but would be nice.
Or maybe it is possible now?

**Comment by mjauvin** [comment, marked as answer]
Maybe add this method to your controller?
```
    public function listExtendModel($model, $definition = null)
    {
        $model->table = "my_list_table";
        return $model;
    }       
```

**Reply by SledgehammerPL** [reply]
Too good to be true ;D

**Reply by SledgehammerPL** [reply]
Looks like it's working... Easy and elegant...

**Reply by SledgehammerPL** [reply]
But what about scopes?

in model i have defined public function scopeMyScope($query)
 {
  $query->addSelect(''10' as ten);
} 

And there model uses $this->table, and I want to use _view instead of $this->table
